lua/dylib.lua

```diff
@@ -30,6 +68,79 @@ int add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
 	enum gk_fib_action action, struct gk_config *gk_conf);
 int del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf);
 
+int convert_ip_to_str(struct ipaddr *ip_addr, char *res, int n);
+
+size_t strlen(const char *s);
+
 ]]
 
 c = ffi.C
+
+-- The following is an example function that can be used as
+-- the callback function of list_fib4_entries() and list_fib6_entries().
+
+-- Parameter fib_dump_entry is going to be released after
+-- print_fib_dump_entry() returns, so don't keep references to fib_dump_entry
+-- or any of the data reachable through its fields.
+
+function print_fib_dump_entry(fib_dump_entry)
+	local i = 0
+	local ret
+	local res = ""
+	local max_ip_addr_len = 128
+	local entry = ffi.cast("struct gk_fib_dump_entry *", fib_dump_entry)
+	local ip_addr = ffi.new("char[?]", max_ip_addr_len)
+	local ip_addr_str
+
+	ret = c.convert_ip_to_str(entry.addr, ip_addr, max_ip_addr_len)
+	if ret < 0 then
+		error("Failed to convert an IP address to string")
+	end
+
+	ip_addr_str = ffi.string(ip_addr, c.strlen(ip_addr))
+	res = res .. "FIB entry for IP prefix: " .. ip_addr_str ..
+		"/" .. entry.len .. " with action " ..
+		tostring(entry.action)
+
+	if entry.action == c.GK_FWD_GRANTOR then
+		ret = c.convert_ip_to_str(entry.grantor_ip,
+			ip_addr, max_ip_addr_len);
+		if ret < 0 then
+			error("Failed to convert an IP address to string")
+		end
+
+		ip_addr_str = ffi.string(ip_addr, c.strlen(ip_addr))
+		res = res .. "\n\tGrantor IP address: " .. ip_addr_str
+	end
+
+	res = res .. "\n\t#Ethernet cache entries: " .. tostring(entry.num_ether_entries)
+
+	local eth_tbl = ffi.cast("struct gk_fib_ether_dump_entry *",
+		entry.ether_tbl)
```

Comment:
Do we really need this cast? Isn't `entry.ether_tbl` already of the type `struct gk_fib_ether_dump_entry *`?

Reply:
Indeed, this cast can be removed.